packages/plugin-core/src/commands/MoveHeader.ts

```diff
@@ -130,21 +147,34 @@ export class MoveHeaderCommand extends BasicCommand<
    * @returns
    */
   private promptForDestination(opts: CommandInput) {
-    const lookupController = LookupControllerV3.create({
+    const lcOpts: LookupControllerV3CreateOpts = {
       nodeType: "note",
-      disableVaultSelection: true,
-    });
+      disableVaultSelection: opts?.useSameVault,
+      vaultSelectCanToggle: false,
+    };
+    const lookupController = LookupControllerV3.create(lcOpts);
     const lookupProvider = new NoteLookupProvider(this.key, {
-      allowNewNote: false,
+      allowNewNote: true,
       noHidePickerOnAccept: false,
     });
+    lookupProvider.registerOnAcceptHook(ProviderAcceptHooks.NewLocationHook);
```

Comment:
lookup by default already gives us the quickpick value, this hooks seems to provide us with a vault if it is not set in the quickpick. instead of a hook, does it make sense to have this as a utility? eg. `PickerUtils.getVaultFromPickerOrEditor`?

Reply:
doh. totally blanked on the fact that you can just grab the vault from the picker.